rkt: fetch images dependencies.

sgotti opened the pull request:
After fetching an image also fetch recursively all its dependencies.
Dependencies needs to be resolved using discovery as the unique useful
informations (as per spec) are the app name and the labels.

Image fetching and dependencies fetching could be done in a single function but
they are left split in two functions as sometimes, for example when importing
images from disk, it'll be useful to disable dependencies fetching.

By now, findImage in rkt/run.go will call fetchImage with withDep = true.

For testing there'll be the need of faking discovery. By now its quite difficult
to fake the meta discovery mechanism. There'll be the need of something like
appc/spec#110

This will probably clash with #509.

jonboulle commented on rkt/fetch.go:
```diff
@@ -344,6 +406,16 @@ func newDiscoveryApp(img string) *discovery.App {
 }
 
+// appToString creates the URL-like image name from the app name and labels
+// TODO(sgotti) move this to appc/spec/discovery?
```
Yeah, that'd be good, the inverse of NewAppFromString

sgotti replied:
Thanks. I'll prepare a PR.

jonboulle commented:
awesome, lgtm, thanks! A test case would be great, though

sgotti commented:
@jonboulle yeah (see the end of the description). I have to think about I way to fake discovery as the httptest server will bound to a local ip and random port without any dns resolution (ex. example.com). I was thinking to wait for the host/port/scheme mapping proposed in appc/spec#110).

jonboulle commented:
@sgotti sorry for not reading :-(

sgotti commented:
@jonboulle This needs a rebase. If it's fine with you I'll wait after #716 to avoid too much cross PR rebasing. 

Should this also wait for something like appc/spec#110 to implements the tests or they can come later?

Otherwise, if you think this has the priority, I'll rebase it now.

jonboulle commented:
@sgotti Sure, let's wait for #716. Do you have thoughts on the discussion in appc/spec#110?

jonboulle commented:
#716 landed so this is ready for rebase

sgotti commented:
@jonboulle I updated the PR. I added a withDeps parameter to the finder/fetcher instead of a function parameter. No tests as per the previous comment.

iaguis commented:
LGTM
